### ISSUE TITLE
Lambda function can't read S3 logs from Kinesis Firehose

### ISSUE THREAD

**Issue by tomjrees** [honeycombio/agentless-integrations-for-aws#22]
I've set up a WAF with logging [per these instructions](https://docs.aws.amazon.com/waf/latest/developerguide/logging.html), with logs being written to an S3 bucket. I'm trying to pipe those logs into Honeycomb.

Kinesis gives you the option to GZIP compress logs, but both settings do not work:

* **Uncompressed**: Logs are stored with `Content-Type: application/octet-stream`
* **Compressed**: Logs are stored `Content-Type: application/octet-stream` and `Content-Encoding: gzip`.

The S3 log parser uses the AWS SDK to download the log file. **The SDK will download uncompressed logs in either case**, because the `Content-Encoding` header means it transparently gets unzipped.

The problematic code is here: https://github.com/honeycombio/agentless-integrations-for-aws/blob/master/s3-handler/main.go#L64

This always crashes with "unable to create gzip reader for object". It's buggy for both settings:

* **Uncompressed**: The `application/octet-stream` header is assumed to mean its a GZIP file. I don't understand why Kinesis Firehose doesn't use a better header, but I think it's wrong to assume it's a GZIP based on this. Crashes on line 72.
* **Compressed**: The `Content-Encoding` header is transparently stripped. We _should_ go to line 65 here and everything _should_ work but the client doesn't get to see the `Content-Encoding` header and will crash on line 72.

Any help you can give will be very gratefully received.

**Comment by zephod** [comment]
Very impressed with the time it took to turn around!

I agree with the approach and think it looks basically fine. I am on the Honeycomb free trial so in order to get started quickly I've written my own Lambda (in Node.js). I've had success so far reading various log types with the following logic:

```javascript
module.exports = async function s3FetchLogFile (Bucket, Key) {
  const params = { Bucket, Key }
  const response = await s3.getObject(params).promise()
  if (response.ContentType === 'application/x-gzip' || response.ContentEncoding === 'gzip') {
    return unzipBuffer(response.Body)
  } else {
    return response.Body.toString('utf8')
  }
}
```

Not sure if your Go version follows the same pattern, but this has successfully been auto-unzipping files with no false positives. Hope this helps.

**Comment by MikeGoldsmith** [comment]
Closing as I believe this was addressed as part of the [v2.0.1](https://github.com/honeycombio/agentless-integrations-for-aws/releases/tag/v2.0.1) release.
